punishments.js

```diff
@@ -603,3 +776,27 @@ Punishments.checkIpBanned = function (connection) {
 
 	return banned;
 };
+
+Punishments.checkRoomBanned = function (user, roomid) {
```

Comment:
And if you pay attention to the rest of this file, you would use `checkNameInRoom` and `checkIpInRoom`

Splitting it into two functions means you can call one function when a new connection merges, and another when a user changes name, and not need to do both checks every time.